gr-blocks/lib/tag_debug_impl.cc

```diff
@@ -57,6 +63,15 @@ int tag_debug_impl::num_tags()
 
 void tag_debug_impl::set_display(bool d) { d_display = d; }
 
+void tag_debug_impl::set_save_all(bool s)
+{
+    if (d_save_all && !s) {
+        d_tags.clear();
```

Comment:
One more thing: Can you hold `d_setlock` while you access `d_tags`. Otherwise, this is not thread-safe.

Reply:
Thanks, @mbr0wn . Tag Debug already had mutex `d_mutex`, which is locked both in `work()` and `current_tags()`. It was missing on `set_save_all()` and I've just added it.